sw/device/rom_exts/rom_ext.c

```diff
@@ -0,0 +1,47 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/rom_exts/rom_ext.h"
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/stdasm.h"
+#include "sw/device/lib/dif/dif_uart.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/print.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"  // Generated.
+
+static dif_uart_t uart;
+
+void rom_ext_boot(void) {
+  dif_uart_result_t init_result = dif_uart_init(
+      (dif_uart_params_t){
+          .base_addr = mmio_region_from_addr(TOP_EARLGREY_UART_BASE_ADDR),
+      },
+      &uart);
+
+  if (init_result != kDifUartOk) {
+    abort();
+  }
+
+  dif_uart_config_result_t config_result =
+      dif_uart_configure(&uart, (dif_uart_config_t){
+                                    .baudrate = kUartBaudrate,
+                                    .clk_freq_hz = kClockFreqPeripheralHz,
+                                    .parity_enable = kDifUartToggleDisabled,
+                                    .parity = kDifUartParityEven,
+                                });
+
+  if (config_result != kDifUartConfigOk) {
+    abort();
+  }
+
+  base_uart_stdout(&uart);
+
+  base_printf("Hello World!\n");
+
+  while (true) {
+    wait_for_interrupt();
+  }
```

Comment:
This seems mildly unnecessary, seeing as this function isn't `noreturn`?

Reply:
This function *is* `noreturn`, look at the header again.

Reply:
At this point I think it probably is a placeholder. The idea behind this change is not have a complete and solid ROM_EXT, but to give us a starting point.

This is a good question, I will add a comment there.

Reply:
Oh, we should probably be clearer about `call` vs `tail` in the `_start.S` functions.

I don't want control flow to return to `_start.S` for either the Mask ROM or the ROM EXTs